hierarchicalforecast/utils.py

```diff
@@ -171,49 +106,41 @@ def aggregate(
 
     Parameters
     ----------
-    df : pandas DataFrame
+    df : pandas or polars DataFrame
         Dataframe with columns `['ds', 'y']` and columns to aggregate.
     spec : list of list of str
         List of levels. Each element of the list should contain a list of columns of `df` to aggregate.
     exog_vars: dictionary of string keys & values that can either be a list of strings or a single string
         keys correspond to column names and the values represent the aggregation(s) that will be applied to each column. Accepted values are those from Pandas aggregation Functions, check the Pandas docs for guidance
-    is_balanced : bool (default=False)
-        Deprecated.
     sparse_s : bool (default=False)
-        Return `S_df` as a sparse dataframe.
+        Return `S_df` as a sparse Pandas dataframe.
 
     Returns
     -------
-    Y_df : pandas DataFrame
+    Y_df : pandas or polars DataFrame
         Hierarchically structured series.
-    S_df : pandas DataFrame
+    S_df : pandas or polars DataFrame
         Summing dataframe.
     tags : dict
         Aggregation indices.
     """
     # Checks
-    if df.isnull().values.any():
-        raise ValueError('`df` contains null values')
-    if is_balanced:
-        warnings.warn(
-            "`is_balanced` is deprecated and will be removed in a future version. "
-            "Don't set this argument to suppress this warning.",
-            category=DeprecationWarning,
-        )
-         
+    for col in df.columns:
+        assert not ufp.is_none(df[col]).any(), f"Column {col} contains null values. Make sure no column in the DataFrame contains null values."            
             
+    if sparse_s and isinstance(df, pl.DataFrame):
+        raise ValueError("Sparse output is not supported for Polars DataFrames.")
+
     # compute aggregations and tags
     spec = sorted(spec, key=len)
     bottom = spec[-1]
     aggs = []
     tags = {}
     # Prepare the aggregation dictionary
-    agg_dict = {
-        "y": ("y", "sum")
-    }
-
-
+    agg_dict = {"y": "sum"}
```

Comment:
This should be fixed; performing the aggregation with ufp doesn't work with a dictionary

Reply:
Fixed